### PR TITLE
SDK: Masks

### PR DESCRIPTION
**Pull request by marcosoliveira91**
Related to: https://github.com/ripe-tech/ripe-core/issues/2290

### CODE REVIEW

File: src/js/visual/config.part.js [hunk 1]
```diff
@@ -588,3 +791,30 @@ ripe.Config.prototype._parseDrag = function() {
...
+
+    var maskContext = mask.getContext("2d");
+    var pixel = maskContext.getImageData(x, y, 1, 1);
+    var r = pixel.data[0];
+    var index = parseInt(r);
+
+    return index;
+};
+
+ripe.Config.prototype._fixEvent = function(event) {
```
Comment (acaldas) [inline src/js/visual/config.part.js]: move this to utils and change to fixEvent

File: src/js/visual/config.part.js [hunk 2]
```diff
@@ -79,7 +97,7 @@ ripe.Config.prototype.update = function(state, options) {
     // if the parts and the position haven't changed
     // since the last frame load then ignores the
     // load request and returns immediately
-    var size = this.element.getAttribute("data-current-size");
+    var size = this.element.dataset.current_size;
```
Comment (hugo-gomes) [inline src/js/visual/config.part.js]: the convention is to still use camel case when accessing `dataset` attributes, or am I missing some gotcha for a particular browser?

File: src/js/visual/config.part.js [hunk 3]
```diff
@@ -588,3 +791,18 @@ ripe.Config.prototype._parseDrag = function() {
     var nextFrame = ripe.getFrameKey(nextView, nextPosition);
     this.changeFrame(nextFrame);
 };
+
+ripe.Config.prototype._getCanvasIndex = function(canvas, x, y) {
+    var canvasRealWidth = canvas.getBoundingClientRect().width;
+    var mask = this.element.querySelector(".mask");
+    var ratio = mask.width / canvasRealWidth;
```
Comment (hugo-gomes) [inline src/js/visual/config.part.js]: let's pray `canvasRealWidth` not to be zero :)

File: src/js/visual/config.part.js (hunk 2 above)
Comment (marcosoliveira91) [reply src/js/visual/config.part.js]: The thing is internally we often use the nomenclature 'data-something_else'.
Previously, on the code, we had all data-attribute names separated by hyphen ('data-current-size') so we updated them to the hyphen/underscore version ('data-current_size'). 

Should we change it?

Comment (marcosoliveira91) [reply src/js/visual/config.part.js]: By the way, when we perform "element.dataset.currenSize" the result will be "data-current-size".

Comment (joamag) [reply src/js/visual/config.part.js]: Just as a side note I was wrong and the `-` character is allowed in the data attributes, but let's just keep using the `_` character instead https://stackoverflow.com/questions/33068777/what-characters-are-allowed-not-allowed-in-the-names-of-custom-data-attribut

**Comment by joamag** [comment]
@marcosoliveira91 please fix the merge conflicts

**Pull request merged**